xbmc/video/windows/GUIWindowVideoNav.cpp

```diff
@@ -73,6 +73,29 @@ using namespace KODI::MESSAGING;
 
 #define CONTROL_UPDATE_LIBRARY    20
 
+namespace
+{
+NodeType GetItemListNodeType(const CFileItemList& items)
+{
+  if (VIDEO::IsVideoDb(items))
+  {
+    return CVideoDatabaseDirectory::GetDirectoryChildType(items.GetPath());
+  }
+
+  if (items.GetContent() == "seasons")
+  {
+    return NodeType::SEASONS;
+  }
+
+  if (items.GetContent() == "episodes")
+  {
+    return NodeType::EPISODES;
+  }
```

Comment:
Database access is expensive. Could it make sense to change the logic here to first check list content, then db?

Reply:
Is IsVideoDb is relatively cheap so I don't think it makes too much difference. The majority case is also likely to be people using the VideoDB not plugins so checking that first makes sense to me.